### PR TITLE
Identify and fix simple using statements in top level expressions

### LINKED ISSUE #68285: IDE0063 doesn't identify "simple using statement" in top-level expressions
**Version Used**: Version 17.7.0 Preview 2.0 [33721.22.main]

## `Program.cs`

```c#
using (var foo = new MyDisposable()) // not suggested here
{
}

class C
{
    void M()
    {
        using (var foo = new MyDisposable()) // suggested here
        {
        }
    }
}
```

I can't see a reason not to suggest it in the top-level context as well.

I'm working on a solution that has a lot of sample code, and therefore a lot of top-level statements. I want to fix up these samples to use modern patterns, and this will need to be done manually.

### PR DESCRIPTION
**Pull request by kandicst**
Fixes #68285

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Thanks. I can review tomorrow!

File: src/Analyzers/CSharp/Analyzers/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -77,9 +78,10 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
         {
             var cancellationToken = context.CancellationToken;
             var outermostUsing = (UsingStatementSyntax)context.Node;
+            var parent = outermostUsing.Parent;
             var semanticModel = context.SemanticModel;
 
-            if (outermostUsing.Parent is not BlockSyntax parentBlock)
+            if (parent is not BlockSyntax && parent is not GlobalStatementSyntax)
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Analyzers/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs:84]:
```suggestion
            if (parent is not (BlockSyntax or GlobalStatementSyntax))
```

File: src/Analyzers/CSharp/Analyzers/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -152,6 +161,21 @@ private static bool DeclaredLocalCausesCollision(ILookup<string, ISymbol> symbol
...
+            SemanticModel semanticModel,
+            SyntaxNode parent,
+            UsingStatementSyntax outermostUsing,
+            UsingStatementSyntax innermostUsing,
+            CancellationToken cancellationToken)
+        {
+            return parent switch
+            {
+                BlockSyntax block => PreservesSemanticsBlock(semanticModel, block, outermostUsing, innermostUsing, cancellationToken),
+                GlobalStatementSyntax global => PreservesSemanticsGlobalStatement(global, semanticModel, cancellationToken),
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Analyzers/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs:173]:
```suggestion
                GlobalStatementSyntax globalStatement => PreservesSemanticsGlobalStatement(globalStatement, semanticModel, cancellationToken),
```
Same here

File: src/Analyzers/CSharp/CodeFixes/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs [hunk 3]
```diff
@@ -88,6 +97,26 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             return currentBlock;
         }
 
+        private static CompilationUnitSyntax RewriteCompilationUnit(CompilationUnitSyntax original, CompilationUnitSyntax current, ISet<SyntaxNode> usingStatements)
+        {
+            var memberToUpdateIndex = original.Members.IndexOf(usingStatements.Contains);
+            var memberToUpdate = (GlobalStatementSyntax)current.Members[memberToUpdateIndex];
+
+            if (memberToUpdate.Statement is UsingStatementSyntax usingStatement &&
+                usingStatement.Declaration != null)
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/CodeFixes/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs:106]:
```suggestion
            if (memberToUpdate.Statement is UsingStatementSyntax { Declaration: not null } usingStatement)
```

File: src/Analyzers/CSharp/Analyzers/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs [hunk 4]
```diff
@@ -98,15 +100,22 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
...
                 return;
 
             // Converting a using-statement to a using-variable-declaration will cause the using's variables to now be
             // pushed up to the parent block's scope. This is also true for any local variables in the innermost using's
             // block. These may then collide with other variables in the block, causing an error.  Check for that and
             // bail if this happens.
+            var variableScopeNode = parent switch
+            {
+                BlockSyntax block => block,
+                GlobalStatementSyntax global => global.Parent!,
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Analyzers/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs:113]:
```suggestion
                GlobalStatementSyntax globalStatement => globalStatement.Parent!,
```
I think this name reads better

**Comment by CyrusNajmabadi** [comment]
Closing due to lack of movement from OP.

**Pull request closed without merging**